pkg/kubernetes/client.go

```diff
@@ -0,0 +1,164 @@
+package kubernetes
+
+import (
+	"k8s.io/api/core/v1"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"errors"
+	"fmt"
+)
+
+var (
+	emptyListOptions = meta_v1.ListOptions{}
+	emptyGetOptions = meta_v1.GetOptions{}
+)
+
+// Client for Kubernetes and Istio APIs
+// It hides the way it queries each API
+type IstioClient struct {
+	k8s *kubernetes.Clientset
+	istio *rest.RESTClient
+}
+
+// NewClient creates a new client to the Kubernetes and Istio APIs.
+// It takes the assumption that Istio is deployed into the cluster.
+// It hides the access to Kubernetes/Openshift credentials.
+// It hides the low level use of the API of Kubernetes and Istio, it should be considered as an implementation detail.
+// It returns an error on any problem.
+func NewClient() (*IstioClient, error) {
```

Comment:
Here too: I'm not sure what's the rest calls triggered here, but some of them can maybe be optimized with goroutines. (but we can leave it for later)

Reply:
Yeah, probably, I didn't wanted to hold this PR for this.
I am looking on some optimizations of the client as well, I will prepare a subtask and use this comment with the other optimizations.